pinot-core/src/main/java/org/apache/pinot/core/common/ObjectSerDeUtils.java

```diff
@@ -156,7 +156,12 @@ public enum ObjectType {
     FloatArrayList(44),
     StringArrayList(45),
     UltraLogLog(46),
-    ThetaSketchAccumulator(47);
+    ThetaSketchAccumulator(47),
+    IntArray(48),
```

Comment:
Can you add more context on the code path requiring ser/de of these arrays? Based on my understanding this class is only used for ser/de the intermediate result of aggregation, so trying to understand why these changes are needed

Reply:
This is also used by `BaseDataTableBuilder.setColumn(...)` which is used by all the agg/groupby/selection queries.

Reply:
Array Literal is pushed down to worker in v2 engine, so need to wire it all the way.

Reply:
We do have `BaseDataTableBuilder.setColumn()` for array types, and it should not go over the general handling of `Object`. Can you double check?

Reply:
You are right, int[], long[], float[], double[] are already implemented. Just need to implement `StringArray`.

Reply:
`String[]` is also supported natively within `DataTableBuilder`. Check the interface

Reply:
The issue is we didn't register the return type for v2 engine for `split` function. So it was `OBJECT` in desired return type.